### DISCUSSION TITLE
Is there a way to edit markdown as plain text?

### DISCUSSION THREAD

**Discussion by adaboese** [Milkdown/milkdown#1232] [Q&A]
I want to allow editing raw markdown in addition to editing with a preview.

Does such option exists?

**Comment by PizzaConsole** [comment]
can you just create your own text area box with the raw markdown in it?

**Comment by matthijskooijman** [comment]
I was looking at the same thing. The online playground has a right sidepanel that has the raw markdown and allows editing it, which makes it look like Milkdown is an editor that can do collaborative editing of both WYSIWYG and raw markdown, which is really what I was looking for.

However, looking at examples (for example I tried https://github.com/datalayer-externals/milkdown-examples/tree/main/vanilla-collab), there is only a WYSIWYG editor. And looking at the playground sources, I see the following code: https://github.com/Milkdown/website/blob/b471e9536aa4e21f2ea70f077e27d19d1e6179c8/src/pages/playground.tsx#L76-L90

There I see that the raw markdown view in the playground is a separate CodeMirror component from the editor, that just pushes its entire contents to the editor when the raw markdown is changed. IIUC, this means that this raw markdown editor is not really collaborative - if two people would use it at the same time, they would just continuously overwrite each other's changes. This would mean that if you do not need collaborative editing, then you can maybe build something based on the codemirror component. If you need collaborative editing, then I *think* it might not be supported (and I have not found any other editor so far that actually does support it).
